File: Python/programmers/etc/level2/N개의_최소공배수.py
def gcd(a,b):
    while b:
        a, b = b, a%b
    return a

def solution(arr):
    ret = 1
    for i in arr:
        ret = ret*i//gcd(ret,i)
    return ret


arr = [2,6,8,14]
print(solution(arr)) # 168

arr = [1,2,3]
print(solution(arr)) # 6

File: Java/programmers/etc/level2/N개의최소공배수.java
package CodeTest.Java.programmers.etc.level2;

public class N개의최소공배수 {
    public int solution(int[] arr) {
        int answer = 1;
        for (int a : arr) {
            answer = (answer*a)/gcd(answer,a);
        }
        return answer;
    }
    
    private static int gcd(int a, int b) {
        while (b != 0) {
            int tmp = a; a = b; b = tmp%b;
        }
        return a;
    }
}
